src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_validate.c

```diff
@@ -51,16 +56,18 @@ int wm_agent_upgrade_validate_status(const char* connection_status) {
 
 int wm_agent_upgrade_validate_system(const char *platform, const char *os_major, const char *os_minor, const char *arch) {
     int invalid_platforms_len = 0;
-    int invalid_platforms_it = 0;
+    int rolling_platforms_len = 0;
+    int platforms_it = 0;
     int return_code = WM_UPGRADE_GLOBAL_DB_FAILURE;
 
     if (platform) {
         if (!strcmp(platform, "windows") || (os_major && arch && (strcmp(platform, "ubuntu") || os_minor))) {
             return_code = WM_UPGRADE_SUCCESS;
-            invalid_platforms_len = sizeof(invalid_platforms) / sizeof(invalid_platforms[0]);
 
-            for(invalid_platforms_it = 0; invalid_platforms_it < invalid_platforms_len; ++invalid_platforms_it) {
-                if(!strcmp(invalid_platforms[invalid_platforms_it], platform)) {
+            // Blacklist for invalid OS platforms
+            invalid_platforms_len = sizeof(invalid_platforms) / sizeof(invalid_platforms[0]);
```

Comment:
Here you can use https://github.com/wazuh/wazuh/blob/959fe420af99e26b30d03bb5b1a7b9bed5815670/src/headers/shared.h#L206